### ISSUE TITLE
Vega usage offline throws error because of schema.

### ISSUE THREAD

**Issue by jgoodall** [vega/vega#3953]
I need to use vega offline, with no internet access, but while the following works:

```
"$schema": "https://vega.github.io/schema/vega/v5.json"
```

If I download the schema and save in the same directory as my json file (`curl -sS https://vega.github.io/schema/vega/v5.json -o schema-v5.json`) and change my vega file to the following it fails:

```
"$schema": "schema-v5.json"
```

The console shows an error: `TypeError: /schema\/([\w-]+)\/([\w\.\-]+)\.json$/g.exec(...) is null`

The schema type is a [URL](https://vega.github.io/vega/docs/types/#URL) so that should be a valid URL.

Not sure what I am doing wrong...

**Comment by domoritz** [comment]
The schema url is not evaluated by Vega I thought. Can you explain what command throws an error? You might need a relative url like ./etc

**Comment by jgoodall** [comment]
I have tried (filename is `schema-v5.json`, in same directory as my `index.html` and vega spec file): 

* `"$schema": "schema-v5.json"`
* `"$schema": "/schema-v5.json"`
* `"$schema": "./schema-v5.json"`
* `"$schema": "http://localhost:8080/vega-schema.json"` (server is running on 8080)

However, I just checked `vega-embed` and found this issue: https://github.com/vega/vega-embed/issues/305

I think that is not working the way I expected, based on that issue. It appears that using `"$schema": "https://vega.github.io/schema/vega/v5.json"` works in an offline environment. I had expected that vega (or vega-embed) would need to use that URL to actually download the file, but that appears not to be the case. It appears that it is only used to determine the version - is that right?

**Comment by domoritz** [comment]
Ah yes, we use the schema url to get the spec type and version. There is a package in the schema repo that does the parsing of urls. We should probably not fail if the schema url does not have the expected format.

**Comment by jgoodall** [comment]
My immediate problem seems to be solved by just using `"$schema": "https://vega.github.io/schema/vega/v5.json"`, so not sure if you want to keep the issue open based on previous comment.

**Comment by domoritz** [comment]
Yes, and file a new issue with Vega schema to not fail on unknown urls.
